### ISSUE TITLE
 [object Object] Issue with Simple Schema and quickForm

### ISSUE THREAD

**Issue by DanMan259** [aldeed/meteor-autoform#1660]
I am not sure why I am getting this error when ever I load in my modal with my quickForm. It just keeps on showing [object Object]. Im assuming that I passed an object and it cant represent it with the label. How can I fix the issue.

HTML users.html

```
<template name="users">
 <div class="row">
        <div class="col s12" align="right">
            <a class="waves-effect waves-light btn modal-trigger" href="#modal1">Add a User</a>
        </div>
        {{#each users}}
            {{>user}}
        {{/each}}
    </div>
    <div id="modal1" class="modal modal-fixed-footer">
        <div class="modal-content">
            <h4>Add new User</h4>
            {{> quickForm collection="Users" id="insertUserForm" type="insert" class="new-user-form"}}
        </div>
        <div class="modal-footer">
            <a href="#!" class="modal-action modal-close waves-effect waves-green btn-flat">
                Close
            </a>
        </div>
    </div>
</template>
```

JavaScript users.js

```
import { Mongo } from 'meteor/mongo';
import SimpleSchema from 'simpl-schema';

SimpleSchema.extendOptions(['autoform']);

export const Users = new Mongo.Collection('users');

const UserSchema = new SimpleSchema({
    firstName: {
        type: String,
        label: "First Name",
    },
    lastName: {
        type: String,
        label: "Last Name",
    },
    email: {
        type: String,
        label: "E-Mail",
    },
    createdAt: {
        type: Date,
        autoform: {
            type: 'hidden',
            label: false,
        },
        defaultValue: new Date(),
    },
});

Users.attachSchema( UserSchema );
```
JavaScript body.js

```
import { Meteor } from 'meteor/meteor';
import { Template } from 'meteor/templating';
import { Users } from '../api/users';

import './body.html';
import './users.html';

window.Users= Users;

Template.body.onCreated(function bodyOnCreated(){
    Meteor.subscribe('users.allusers');
});

Template.registerHelper('formatDate', function(date){
    return moment(date).format('ll');
});

Template.users.helpers({
    users(){
        return Users.find({});
    },
});

Template.users.onRendered(function(){
    $('#modal1').modal();
});
```
![screenshot from 2018-05-07 12-44-40](https://user-images.githubusercontent.com/34361062/39713401-a852b966-51f4-11e8-87db-d9c04fd19177.png)

**Comment by marcoschwartz** [comment]
Same issue here, for me it is coming from the latest version of simpl-schema (1.5.0). When I revert to 1.4.3 the issue goes away. I noticed it's only with quickForms, when I use #autoform it works without issues.

**Comment by erisevaline** [comment]
Exactly the same issue for me. Hope we're gonna get answers :)

**Comment by jankapunkt** [comment]
Can confirm this when using `simpl-schema@1.5.0` while different AutoForm versions did not affect this.

Edit: Mine is also working with `simpl-schema@1.4.3`. It seems that there are compatibility issues since 1.5.0 as we can see also in #1662 #1654

**Comment by rootedsoftware** [comment]
Duplicate of #1653

**Comment by rikyperdana** [comment]
With Users.find pass the whole cursor to it.
For the array, do this instead
return Users.find({}).fetch();

**Comment by cheinle** [comment]
Hi I have the same trouble.  However, I do not understand how to correctly go from simpl-schema (1.5.0) to simpl-schema (1.4.3). I have tried deleting simpl-schema from the node modules directory and then updating package.json to "simpl-schema": "^1.4.3" .  However, if I run meteor npm install, it says everything is up to date and does not install anything. If i run meteor npm install --save simpl-schema it installs 1.5.0 again. Any help would be greatly appreciated as I have been struggling with this for awhile. Thanks ahead of time!

**Comment by jankapunkt** [comment]
@cheinle 

```
meteor npm uninstall --save simpl-schema
meteor npm prune
meteor npm install --save simpl-schema@1.4.3
```

should work

**Comment by cheinle** [comment]
@jankapunkt Thanks so much! It did work and the downgrade also fixed my trouble.

**Comment by abate** [comment]
I've noticed the same problem when the autoform type for an array is "hidden" .

**Comment by MrLexJR** [comment]
Thanks so much ✌️  @jankapunkt

**Comment by mikeacre** [comment]
@jankapunkt you are the best!

**Comment by mikeacre** [comment]
Exception in template helper: TypeError: Cannot read property 'singleType' of undefined
at SimpleSchema.getQuickTypeForKey  PAsting this here so when people search for the error it shows up.

**Comment by jankapunkt** [comment]
Fix (in SimpleSchema) is ready to be merged: https://github.com/aldeed/simple-schema-js/pull/291

**Comment by thoni56** [comment]
Going back from simpl-schema@1.5.3 to 1.4.3 fixed my issue with [object Object] in quickform also.

**Comment by mikeacre** [comment]
Getting this error again with a update-pushArray
{{> quickForm collection="Contacts" doc=contact._id id="newNote" type="update-pushArray" scope="contacts"}}

**Comment by jankapunkt** [comment]
Should be fixed in the latest impl-schema, see #1654 and https://github.com/aldeed/simple-schema-js/pull/306

**Comment by jankapunkt** [comment]
This is resolved
